### ISSUE TITLE
No error reported for duplicate `ref` in lambda signature

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#62085]
No errors are reported for lambda signatures below:
```csharp
delegate void D(ref int i);

class Program
{
    static void Main()
    {
        D d1 = (ref ref int i) => { };
        D d2 = (in ref int i) => { };
        D d3 = (out ref int i) => { };
    }
}
```

**Comment by Youssef1313** [comment]
The error for other symbols come from https://github.com/dotnet/roslyn/blob/69bdfb29ce4d027c5f5bf70f50be9bebcb5df1aa/src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs#L19

But it looks like lambdas have their own `MakeParameters`:

https://github.com/dotnet/roslyn/blob/220b3ebfd25a839b3abc1b0096eaca0b9de0a3c2/src/Compilers/CSharp/Portable/Symbols/Source/LambdaSymbol.cs#L303

**Comment by cston** [comment]
> But it looks like lambdas have their own `MakeParameters`:

I'm changing `Binder.AnalyzeAnonymousFunction()` to use `ParameterHelpers.CheckModifiers()` and `ParameterHelpers.GetModifiers()`.

**Comment by cston** [comment]
Fixed in https://github.com/dotnet/roslyn/pull/62155.
